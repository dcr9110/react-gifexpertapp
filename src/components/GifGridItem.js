import React from 'react';
//impt
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card animate__animated animate__jello">
            <p>{ title } </p>
            <img src={ url } alt={ title } />           
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}