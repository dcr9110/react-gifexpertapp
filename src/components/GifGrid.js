import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    const { data:images, loading } = useFetchGifs( category );  

    return (
        <>
            <h3> { category }</h3>
              {loading && <h2 className='animate__animated animate__backInDown'>Cargando...</h2> }
            <div className="card-grid">
                {
                    images.map( img =>(
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            </div>
         </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}