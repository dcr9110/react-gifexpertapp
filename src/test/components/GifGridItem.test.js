import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
const { shallow } = require("enzyme");

describe('Pruebas en <GifGridItem/>', ()=>{

    const title = 'un titulo';
    const url = "https://localhost/hola.jpg";
    const wrapper = shallow( <GifGridItem   
        title = {title} 
        url = {url}
        /> );
 
    test('debe de mostrar el componente correctamente',()=>{

        expect( wrapper ).toMatchSnapshot();
  
    })

    test('debe tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title ); 

    })
    

    test('should ', () => {
        
    })
    
})