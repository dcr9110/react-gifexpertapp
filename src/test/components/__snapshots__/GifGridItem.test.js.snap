// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__jello"
>
  <p>
    un titulo
     
  </p>
  <img
    alt="un titulo"
    src="https://localhost/hola.jpg"
  />
</div>
`;
